official/utils/misc/distribution_utils.py

```diff
@@ -96,3 +98,67 @@ def per_device_batch_size(batch_size, num_gpus):
           ).format(num_gpus, batch_size, batch_size - remainder)
     raise ValueError(err)
   return int(batch_size / num_gpus)
+
+# The `SyntheticDataset` is a temporary solution for generating synthetic data
+# directly on devices. We will have better support in `tf.data` or Distribution
+# Strategy later.
```

Comment:
Also add a comment saying this code path only affects using distribution strategy with keras.

Reply:
Done.